pages/api/internal/assets/[anchor]/[assetType]/index.js

```diff
@@ -140,17 +146,31 @@ export default async function handle(req, res) {
 
 		if (existingAsset) {
 			await connectWithExistingAsset(existingAsset, nft)
-
+			rmSync(uploadedFile.path)
 			return res.json({
 				existingAsset
 			})
 		} else {
 			try {
-				writeFileSync(path.join(STORAGE_DIR, fileHash), fileContent)
+				// Split files per Smart contract. 
+				// This is a logical separation and also reduces the risk of Filesystem-overload
+				// due to too many files in the same directory
+				const storageFileName = path.join(nft.contract.csn, fileHash)
+				const targetFile = path.join(STORAGE_DIR, storageFileName)
+				const targetDir =path.dirname(targetFile)
+
+				// Create contract-directory in case it does not exist
+				if(!existsSync(targetDir)) {
```

Comment:
I think moving this whole process into its own method would be a bit cleaner, since this handler is getting too big already as it is. It is a minor adjustment though

Reply:
Noted for later: https://github.com/authenticvision/digitalsoul/issues/42